### PR TITLE
Backport various AWS fixes from upstream

### PR DESCRIPTION
**Pull request by gnufied**
Fixes following issues:

1. https://trello.com/c/aR8Lor1h/573-2-ebs-aws-optimization-for-api-quota
2. https://trello.com/c/QyGwUOU8/590-8-fix-dangling-volumes-attached-to-nodes, which is a source of many known bugs
3. https://bugzilla.redhat.com/show_bug.cgi?id=1463844

cc @jsafrane @childsb @eparis

### CODE REVIEW

File: vendor/k8s.io/kubernetes/pkg/cloudprovider/providers/aws/aws.go [hunk 1]
```diff
@@ -1499,6 +1522,28 @@ func (d *awsDisk) describeVolume() (*ec2.Volume, error) {
...
+		glog.Errorf("Error fetching node %s with %v", nodeName, fetchErr)
+		return
+	}
+
+	taint := &v1.Taint{
+		Key:    nodeWithImpairedVolumes,
+		Value:  "true",
+		Effect: v1.TaintEffectNoSchedule,
+	}
+	err := controller.AddOrUpdateTaintOnNode(c.kubeClient, string(nodeName), taint)
```
Comment (jsafrane) [inline vendor/k8s.io/kubernetes/pkg/cloudprovider/providers/aws/aws.go]: Is "cloud-provider" allowed to get and update Node objects?

Comment (gnufied) [reply vendor/k8s.io/kubernetes/pkg/cloudprovider/providers/aws/aws.go]: Good catch. I have opened a follow up PR for fixing cloud-provider's access to node objects. https://github.com/kubernetes/kubernetes/pull/56709 

Originally when I tested this, I was running a cluster without RBAC. :(

**Comment by gnufied** [comment]
@markturansky @jwforres This PR changes list of available serviceaccounts. It introduces a new serviceaccount called "system:aws-cloud-provider". The related test in `web_console_role_test.go` started to fail after this change and I had to update list of roles the test assumes.

**Comment by jwforres** [comment]
@benjaminapetersen do we need that role test anymore since we reverted to using the whitelist?

**Comment by gnufied** [comment]
@jsafrane @eparis @childsb can I get review/lgtm/approve please?

**Comment by benjaminapetersen** [comment]
@jwforres nope, we have an issue created to remove the roles test #17257

**Comment by rootfs** [comment]
given aws a clusterole is something similar to https://github.com/kubernetes/kubernetes/pull/53251/files

cloud providers in k8s don't get cluster role (yet)

@pospispa @deads2k @liggitt

**Comment by liggitt** [comment]
upstreams are held for the rebase onto 1.9 alpha

**Comment by pospispa** [comment]
I don't have anything against giving aws a cluster role.

My use case in https://github.com/kubernetes/kubernetes/pull/53251/files was different. I wanted to allow an external provisioner to get K8s cluster topology/nodes in order to choose a zone in which a PV will be provisioned. However, AFAIK, CSI gives the topology to a provisioner as one of the parameters when it asks the provisioner to provision a PV. So my use case should be solved by CSI.

**Comment by gnufied** [comment]
@rootfs the securityaccount being created in this PR is unrelated and I think pretty self-sufficient. It was agreed in upstream PR to allow the cloudprovider account.

**Comment by gnufied** [comment]
[](url)

**Pull request closed without merging**